### DISCUSSION TITLE
Plotting grains with certain orientations in color map

### DISCUSSION THREAD

**Discussion by kopparthiravikiran** [mtex-toolbox/mtex#1773] [Ask Anything]
I am trying to plot the GOS map in MTEX. In the colour map, using the below code, I can plot grains with all orientations (0 to 4.5°). However, I would like to colour grains whose orientation is beyond 2° as white. Could anyone please suggest to me what is the code for that?

Thank you very much.

%%
 
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'));
ebsd(grains(grains.grainSize < 5)) = [];
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',5*degree);
ebsd = ebsd.project2FundamentalRegion;
grains = smooth(grains,5);
 
% plot the data
plot(ebsd, ebsd.orientations)
hold on
plot(grains.boundary,'lineWidth',2)
hold off
 
%%
 
GOS = ebsd.grainMean(mis2mean.angle);
 
plot(ebsd('indexed'),colorKey.orientation2color(ebsd('indexed').orientations))

% plot it
plot(grains, GOS ./ degree)
mtexColorbar('title','')
setColorRange([0,4.5])

**Comment by kilir** [comment]
Hi,
in order to plot grains which fulfill a certain condition such as yours with a constant color you can do:
```
plot(grains(condition),'FaceColor','WhiteSmoke','linecolor','k'),
```
Similarlarly, you can plot those which belong to ~condition differently.
Cheers,
Rüdiger

**Reply by kopparthiravikiran** [reply]
Thank you very much. This is exactly what I am looking for.

Regards,
Ravi.
